R/experimentSpec.R

```diff
@@ -254,28 +255,26 @@ experimentSpecServer <- function(id,
       name <- input$name
       req(name)
 
-      filtered_mae <- datasets$get_filtered_dataset(mae_name)
-      filter_states <- filtered_mae$get_filter_states(name)
-      subset_queue <- filter_states$queue_get("subset")
-      sapply(subset_queue, function(x) x$get_call())
```

Comment:
this is a good change

Reply:
does this not affect functionality? will be good to run the shiny test and check that... I remember we specifically checked for the filter code here

Reply:
@danielinteractive is there any reason you checked the code specifically?

Reply:
In my opinion checking filter-code is a responsibility of teal tests not tmh

Reply:
@danielinteractive we test the filter-calls in several configurations in `teal.slice`, so there is no need to test them again in tmh

https://github.com/insightsengineering/teal.slice/blob/80aa3c2c6943fb41f5204edd0ff61a07b123c3c0/tests/testthat/test-MAEFilteredDataset.R#L121

https://github.com/insightsengineering/teal.slice/blob/80aa3c2c6943fb41f5204edd0ff61a07b123c3c0/tests/testthat/test-MAEFilteredDataset.R#L230

https://github.com/insightsengineering/teal.slice/blob/80aa3c2c6943fb41f5204edd0ff61a07b123c3c0/tests/testthat/test-MAEFilteredDataset.R#L301

Reply:
I mean the idea was to trigger a change based on the subset code change in teal. maybe that is now handled by `teal.slice` already. I think as long as functionality does not change it is ok

Reply:
subset code is a consequence of the change in the "subset" filter-states. It's enough to observe if the state changed

Reply:
perfect, thanks @gogonzo for confirming!

Reply:
Thanks @danielinteractive 
Just FYI - we will merge this PR to the feature branch and we will continue working there until we fix everything - this PR has `main` as base branch but only to have a integration tests running, eventually we will switch to feature branch and this won't be included in the nearest release.

Reply:
ah ok cool thanks for info